src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -0,0 +1,116 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Internal.TypeSystem;
+using Internal.IL;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem.Interop
+{
+    public static class MarshalHelpers
+    {
+        /// <summary>
+        /// Returns true if this is a type that doesn't require marshalling.
+        /// </summary>
+        public static bool IsBlittableType(TypeDesc type)
+        {
+            type = type.UnderlyingType;
+
+            if (type.IsValueType)
```

Comment:
Note there are also class case - it can be blittable by itself if it is sequential/explicit and only has such fields, so that we could pin the class if needed. But in nested case we need to ignore such classes. See HasLayoutMetadata and CollectLayoutFieldMetadataThrowing in https://github.com/dotnet/coreclr/blob/master/src/vm/fieldmarshaler.cpp
However, given the complexity involved, we could choose to ignore such case until there is a need.